packages/framework/src/client.ts

```diff
@@ -86,6 +89,7 @@ export class Client {
       apiUrl: resolveApiUrl(providedOptions?.apiUrl),
       secretKey: resolveSecretKey(providedOptions?.secretKey),
       strictAuthentication: !isRuntimeInDevelopment(),
+      compileControls: providedOptions?.compileControls ?? true,
```

Comment:
I wonder if compileControls is the most explaining terminology of what actually happens. As Compile can mean a lot of things 🤔 should we be more direct here: parseLiquidControls or smth? I do remember that we wanted to allow to replace the liquid with other custom parsers tho

Reply:
Valid point. This was the most fragile change in the i n this PR. As we think of the future evolution of this package, it’s essential to design it properly.  

I wouldn’t go with `parseLiquidControls`, as it’s too specific. As you mentioned, other libraries could handle control compilation, so we need a more generic approach.

Reply:
This PR has a performance improvement and an option to control parsing. I believe the framework needs to be opinionated when it comes to Liquid parsing. This shouldn't be an option but we should proceed with the fix to parse only when necessary in specific actions.

Reply:
I provided a detailed answer in the [comment](https://github.com/novuhq/novu/pull/7590/files#r1943424081) above let's continue the discussion there :)